gufe/protocols/protocol.py

```diff
@@ -124,10 +124,11 @@ def _defaults(self):
         return {}
 
     def _to_dict(self):
-        return {'settings': self.settings}
+        return {'settings': self.settings.json()}
 
     @classmethod
     def _from_dict(cls, dct: Dict):
+        # TODO: We need to deserialize from a json, not a dict
```

Comment:
You should still be receiving a dict, but it is a dict with a single key `settings`, which maps to a JSON string. The stuff in openff-models should have a way to reload from that JSON string; @mikemhenry can probably tell you details.

Should look something like:

```python
@classmethod
def _from_dict(cls, dct: Dict):
    settings = some_openff_models_magic(dct['settings'])
    return cls(settings=settings)
```

Reply:
It should be as "simple" as using something like `parse_raw` https://pydantic-docs.helpmanual.io/usage/models/#helper-functions to turn the json string back into a settings object. I know this works since that is how I setup the test.